server/src/main/java/org/elasticsearch/cluster/SnapshotsInProgress.java

```diff
@@ -343,14 +343,21 @@ private static boolean assertConsistentEntries(Map<String, ByRepo> entries) {
                 assert entry.repository().equals(repository) : "mismatched repository " + entry + " tracked under " + repository;
                 for (Map.Entry<RepositoryShardId, ShardSnapshotStatus> shard : entry.shardsByRepoShardId().entrySet()) {
                     final RepositoryShardId sid = shard.getKey();
+                    final ShardSnapshotStatus shardSnapshotStatus = shard.getValue();
                     assert assertShardStateConsistent(
                         entriesForRepository,
                         assignedShards,
                         queuedShards,
                         sid.indexName(),
                         sid.shardId(),
-                        shard.getValue()
+                        shardSnapshotStatus
                     );
+
+                    // TODO independent PR to add this assertion
```

Comment:
I think this comment is no longer necessary
```suggestion
```

Reply:
Oops, thanks :) e2e660d3c2cc80579e6d30269c22f5a2d9fc2aab